### ISSUE TITLE
Render callback always gets props

### ISSUE THREAD

**Issue by fr1n63** [relay-tools/react-router-relay#142]
I've been playing with the 0.13.0-rc2 and I can happily solve all my earlier issues so all good thanks...

However I am having to filter the router props out of the props that are supplied to the components.

The example given has `render={({ props }) => props ? <Widget {...props} /> : <Loading />}`

However this always gets given an object with these router properties 

```
  'children',
  'history',
  'location',
  'params',
  'route',
  'routeParams',
  'routes'
```

...so `<Widget>` is always rendered.

I think this is an issue in RouteContainer where you always combine the routerProps and props

```
element = render.call(route, {
      ...renderArgs,
      props: { ...routerProps, ...props },
    });
```

Whereas Relay ReadyStateRenderer had a conditional to determine if you should get props at all.

```
props: readyState.ready ?
            this.state.getContainerProps(this.props) :
            null,
```

Like i said, I can filter the router props out and make a decision on that, but it does break the example and feels like a hack on my end.

**Comment by taion** [comment]
Nope, that's a bug on my end. Note that it works correctly in the other code path.

**Comment by fr1n63** [comment]
Cool, thanks

**Comment by taion** [comment]
Trickier than I thought. You always want the router props in the render method, after all.

**Comment by fr1n63** [comment]
Ah, never as straight forward as it seems...

Could you add a bit of logic into RouteContainer and pass a flag into the render callback...?

```
const relayRenderArgs = Object.assign({
  hasRelayProps: Object.keys(props).length > 0
}, renderArgs);

element = render.call(route, {
      ...relayRenderArgs,
      props: { ...routerProps, ...props },
    });
```

**Comment by taion** [comment]
I'm going to pass in a separate `routerProps` with the router props, to keep the semantics around `props` the same.

**Comment by taion** [comment]
https://github.com/relay-tools/react-router-relay/pull/143

**Comment by fr1n63** [comment]
That works too :)

**Comment by taion** [comment]
Released as v0.13.0-rc.4.

**Comment by fr1n63** [comment]
Perfect, all sorted thanks.
